Bottom: 6e789c92ebdef91b404ee5b81a59493838f588f3
Top:    6b9a5330ec3dc877c0aed7048687b42265b48247
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-05-10 23:03:24 +0100

Temp fixes and diagnostics


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 6a3862f..7555599 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1160,7 +1160,11 @@ label_contour (PLINE * a)
       assert (label == INSIDE || label == OUTSIDE);
       LABEL_NODE (cur, label);
     }
-  while ((cur = cur->next) != first_labelled);
+  while ((cur = cur->next) != first_labelled &&
+	 !(cur->next == &a->head && first_labelled == NULL));
+
+  if (first_labelled == NULL)
+    fprintf (stderr, "OOPS, DIDN'T LABEL ANYTHING!\n");
 #ifdef DEBUG_ALL_LABELS
   print_labels (a);
   DEBUGP ("\n\n");
diff --git a/src/sweep.h b/src/sweep.h
new file mode 100644
index 0000000..e69de29
